File: 0547-number-of-provinces/0547-number-of-provinces.cpp
#include <bits/stdc++.h>
using namespace std;

void dfs(int i, vector<int> &visited, vector<vector<int>> &isConnected) {
   visited[i] = 1;
  for(int num = 0; num < isConnected[i].size(); ++num) {
    if(isConnected[i][num] == 1 && !visited[num]) {
        dfs(num, visited, isConnected);
    }
}

   }

class Solution {
public:
    int findCircleNum(vector<vector<int>>& isConnected) {
        int n = isConnected.size();
        vector<int> visited(n, 0);
        int count = 0;

        for (int i = 0; i < n; ++i) {
            if (!visited[i]) {
                ++count;
                dfs(i, visited,isConnected);
            }
        }

        return count;
    }
}; 

File: 0547-number-of-provinces/0547-number-of-provinces.java
class Solution {
    public void dfs(int i,int [][] grid,boolean[] visited) {
        visited[i] = true;
        for(int x=0; x<grid[i].length; x++) {
            if(!visited[x] && grid[i][x] == 1) {
                dfs(x,grid,visited);
            }
        }
    }
    public int findCircleNum(int[][] grid) {
        int n = grid.length;
        boolean visited[] = new boolean[n];
        int count = 0;
        for(int i=0; i<n; i++) {
            if(!visited[i] ) {
                ++count;
                dfs(i,grid,visited);
            }
        }
        return count;
    }
}